SimPEG/Directives.py

```diff
@@ -708,82 +735,88 @@ def validate(self, directiveList):
         dList = directiveList.dList
         self_ind = dList.index(self)
         lin_precond_ind = [
-            isinstance(d, Update_lin_PreCond) for d in dList
+            isinstance(d, UpdateJacobiPrecond) for d in dList
         ]
 
         if any(lin_precond_ind):
-            assert(lin_precond_ind.index(True) > self_ind), (
-                "The directive 'Update_lin_PreCond' must be after Update_IRLS "
+            assert lin_precond_ind.index(True) > self_ind, (
+                "The directive 'UpdateJacobiPrecond' must be after Update_IRLS "
                 "in the directiveList"
             )
         else:
             warnings.warn(
                 "Without a Linear preconditioner, convergence may be slow. "
-                "Consider adding `Directives.Update_lin_PreCond` to your "
+                "Consider adding `Directives.UpdateJacobiPrecond` to your "
                 "directives list"
             )
         return True
 
 
-class Update_lin_PreCond(InversionDirective):
+class UpdateJacobiPrecond(InversionDirective):
```

Comment:
I think we could just spell this out `UpdateJacobiPreconditioner`

Reply:
On master it is now ```UpdatePreconditioner``` , having in mind that the function could eventually take different PC styles. Are you good with this?